routes/authentication_cr.js

```diff
@@ -0,0 +1,81 @@
+const express = require("express");
+const router = express.Router()
+const bcrypt = require("bcryptjs")
+
+//importing local files
+const db = require("../db/models");
+const { environment } = require("../config");
+const { asyncHandler, csrfProtection, signUpValidator, validationResult, loginValidators } = require("./utiles");
+
+//Sing-Up
+
+router.post("/sign-up", csrfProtection, signUpValidator, asyncHandler(async(req, res) => {
+
+    const { username, email, password } = req.body
+
+    const validatorError = validationResult(req)
+    if(validatorError.isEmpty()) {
+        const hashedPassword = await bcrypt.hash(password, 10);
+        await db.User.create({
+            username,
+            email,
+            password: hashedPassword,
+            createdAt: new Date(),
+            updatedAt: new Date()
+        });
+
+
+        res.json({}); // Authentication Token to be inserted in the future
+
+
+    } else {
+        const errors = validatorError.array().map((error) => error.msg);
+        res.json({ errors });
+    }
+}))
+
+router.get("/demo", asyncHandler( async (req, res) => {
+    const user = await db.User.findOne({
+        where: {
+            username: "DemoUser",
+            email: "DemoUser@NatlRoutes.com"
+        }
+    });
+    console.log(user);
+    if (!user) {
+        res.send("Please seed all files");
+    } else {
+        req.session.auth = {userId: user.id, username: user.username};
+        res.redirect('/my-routes');
+    }
+}))
+
+router.get("/logout", asyncHandler( async (req, res) => {
+    delete req.session.auth;
+    res.redirect("/");
+}))
+//Login
+
+router.post("/login",
+    csrfProtection,
+    loginValidators,
+    asyncHandler (async(req, res) => {
```

Comment:
I'm a little confused about what this route is supposed to be doing! It seems like the route to log a user in, but its unconditionally rendering a template on line 76, but if there are validation errors, it will res.json `errors`, but also line 76 will still run anyway which means two responses will be generated. This function doesn't seem like a finished work.

Reply:
Already fixed this one, I believe😁